core/vtk/ttkDistanceField/ttkDistanceField.h

```diff
@@ -94,20 +70,20 @@ class TTKDISTANCEFIELD_EXPORT ttkDistanceField : public vtkDataSetAlgorithm,
   ttkDistanceField();
   ~ttkDistanceField() override;
 
-  TTK_SETUP();
-
   int FillInputPortInformation(int port, vtkInformation *info) override;
+  int FillOutputPortInformation(int port, vtkInformation *info) override;
+  int RequestData(vtkInformation *request,
+                  vtkInformationVector **inputVector,
+                  vtkInformationVector *outputVector) override;
 
 private:
-  std::string ScalarField;
-  int OutputScalarFieldType;
-  std::string OutputScalarFieldName;
-  bool ForceInputVertexScalarField;
-  std::string InputVertexScalarFieldName;
+  // std::string ScalarField{"ScalarField"};
+  int OutputScalarFieldType{DistanceType::Float};
+  std::string OutputScalarFieldName{"DistanceFieldValues"};
+  bool ForceInputVertexScalarField{false};
+  std::string InputVertexScalarFieldName{"InputScalarField"};
```

Comment:
this should be replaced with optional array retrieval, see core/vtk/ttkIntegralLines/ for an example.
this module has a nearly identical use-case, you can use it as an example in many aspects.

Reply:
I'm not sure if I've resolved this one because I'm not sure if OutputScalarFieldType and OutputScalarFieldName can be replaced with optional array retrieval.